common/src/main/java/com/wynntils/models/items/items/game/GearItem.java

```diff
@@ -33,13 +37,23 @@ public class GearItem extends GameItem
                 PowderedItemProperty,
                 RerollableItemProperty,
                 ShinyItemProperty,
-                IdentifiableItemProperty<GearInfo, GearInstance> {
+                IdentifiableItemProperty<GearInfo, GearInstance>,
+                SetItemProperty,
+                RequirementItemProperty {
     private final GearInfo gearInfo;
     private final GearInstance gearInstance;
-
-    public GearItem(GearInfo gearInfo, GearInstance gearInstance) {
+    private final Optional<SetInfo> setInfo;
```

Comment:
Let's address `SetInfo` first. You should be able to put this in `GearInfo`, load gear data serially, just like I mentioned before. That way, you won't have a race when trying to get info from a model that may be still loading.